Automattic/sensei
# Integrate status controls with the progress bar

### Changes proposed in this Pull Request

* Synchronises the module status controls with the lessons control and with the progress bar.
* I added a new store to handle the state updates, `sensei/course-status`.
* I tried to avoid using the outline structure as it would make the solution complicated. Instead I used a single array with the completed lessons as the source of truth for everything.
* The downside is that we need to calculate the module status on the fly.
* I also added a `RangeControl` for the progress bar as I imagined that it would be tedious for the user to set the module/lessons status to check how the bar looks while styling. The value of the progress is not stored so it is going to be lost on refreshing the page and if a value is set it overrides the progress coming from the lessons. The numbers on the title are going to still be correct though. We can remove it if we don't think that it adds value.

### Testing instructions

* Add the outline block and the progress bar.
* Play around with the lesson and module status controls and observe the progress bar changing.
* Check that the module status and lesson statuses are sychronised.
